logstreams/src/main/java/io/zeebe/logstreams/spi/LogStorageReader.java

```diff
@@ -11,11 +11,14 @@
 import org.agrona.DirectBuffer;
 
 public interface LogStorageReader extends Closeable {
+
   /**
-   * Returns the address of the first block in the storage or {@link
-   * LogStorage#OP_RESULT_INVALID_ADDR} if the storage is empty.
+   * Returns true if nothing could be read from the log, regardless of the current position of the
+   * reader - meaning any seek operations would not change the result of this call.
+   *
+   * @return true no records are readable, false otherwise
    */
-  long getFirstBlockAddress();
+  boolean isEmpty();
```

Comment:
Should `isEmpty` be part of the `LogStorage`? And when this returns `true` `hasNext` should return false right? Or either read return `NO_DATA`.

Reply:
What do you propose? The reader needs to figure out somehow if the underlying log is empty; before it was using `getFirstBlockAddress` and checking if it was negative, which imo is kind of hacky, when the goal is to determine if the underlying storage is empty.

Reply:
Was not sure whether isEmpty is the right place in the reader. Normally isEmpty are part of collections or persistence. And the reader or iterators has normally an hasNext method. This is more an abstraction/interface design thing I wanted to highlight/ask.

Reply:
Agreed, it should be in the `LogStorage` (though it's arguable if our reader interfaces are really iterators since they do random access as well). In general I'd like to revise the log stream abstraction anyway, which apparently is an OKR next quarter. As it is, for now, I can move it to `LogStorage` and query the storage directly from the log reader (though I wanted to avoid having to keep an extra reference =/).